### ISSUE TITLE
Sample (stochastically) from the model?

### ISSUE THREAD

**Issue by thisiscam** [zhouhaoyi/Informer2020#40]
Hi,

I was wondering if it is possible to sample from the Informer model? 

To be more specific:
During decoding in classic Transformer, each token follows a categorical distribution (i.e. the softmax) and is sampled/generated one by one. I understand that it's an advantage of Informer that this sequential decoding is not needed anymore. 
But does this mean that one cannot get diverse samples from the Informer model? 
If not, how would one go about generating those samples?

Thanks!

**Comment by cookieminions** [comment]
Hi,

Informer is designed for time series forecasting, so each prediction value is a number instead of a categorical distribution. If you want to get diverse samples, you can follow the same processing methods in NLP translation or generation tasks, applying the processing of each decoding step to the final generated result of Informer.

**Comment by thisiscam** [comment]
> applying the processing of each decoding step to the final generated result of Informer.

I don't think I fully understand this. Perhaps you are suggesting using Informer with a sequential decoding process?

**Comment by cookieminions** [comment]
In the original decoding process, the distribution of `one point` is obtained at each step. If you use Informer, maybe you can get the complete prediction sequence by using Informer first, and then get the distribution of `all points` for the entire sequence.

**Comment by thisiscam** [comment]
>  and then get the distribution of all points for the entire sequence

Hmm, I'm not sure what distribution Informer imposes on the output sequence, if at all? 
I view Informer as predicting the MLE of `all points`, is that view correct? 

If you'd like, I can move this discussion to email. Thanks!

**Comment by cookieminions** [comment]
At present, Informer makes numerical predictions for each point, and does not impose any distribution. I think that predicting the MLE of all points can be discussed and implemented. You can try to use Informer to generate the `distribution sequence` or calculate the distribution separately for each point in the generated prediction sequence. And you may need to modify the loss function.

Welcome to discuss with us via email.

**Comment by zhouhaoyi** [comment]
Suppose there is no more discussion. I will close now.
